spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/CoreTest.java

```diff
@@ -17,63 +17,59 @@
 package org.springframework.cloud.kubernetes.fabric8.config;
 
 import java.util.HashMap;
+import java.util.Map;
 
-import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
-import io.fabric8.kubernetes.api.model.SecretBuilder;
 import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.runner.RunWith;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.core.env.Environment;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
 @SpringBootTest(classes = TestApplication.class,
 		properties = { "spring.application.name=testapp", "spring.cloud.kubernetes.client.namespace=testns",
 				"spring.cloud.kubernetes.client.trustCerts=true", "spring.cloud.kubernetes.config.namespace=testns",
 				"spring.cloud.kubernetes.secrets.enableApi=true" })
+@EnableKubernetesMockClient(crud = true, https = false)
 public class CoreTest {
 
-	@ClassRule
-	public static KubernetesServer mockServer = new KubernetesServer();
-
 	private static KubernetesClient mockClient;
 
 	@Autowired
 	private Environment environment;
 
-	@Autowired(required = false)
+	@Autowired
 	private Config config;
 
-	@Autowired(required = false)
-	private KubernetesClient client;
+	// not a fan of changing the type from KubernetesClient, but because of:
```

Comment:
this was already fixed by me, waiting for a review. I've also asked if this will be back-ported. If it is (and you are OK with that, of course), I will upgrade the fabric8 libraries and revert this back.

Reply:
What version of fabric8 was it changed in?  Maybe we should make that change first, before making these?

Reply:
this is still pending review from my PR [here](https://github.com/fabric8io/kubernetes-client/pull/3151). The good news is that it is targeted in `4.13.4` and not `5.x.x`. I am more than OK to wait for `4.13.4` release and only then merge this one.

That being said - I do have a minor question here. Why isn't this project moving towards fabric8 - `5.x.x`? thank you

Reply:
OK cool, its up to you.  If you don't mind going back and making that change I am happy to merge this now.

We will move to 5.x.x in our next release train.  We just don't want to make a major change like that and cause it to break people in a service release.

Reply:
I will make the change in a separate PR, as soon as `4.13.4` is released. pinky-promise!